root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -1787,3 +1776,118 @@ test('Cleanup', function (t) {
+                    tested = true;
+                } else {
+                    st.fail('Test is invalid: Input entity type "i_ent" is specified without expected relationship type "match": ' + JSON.stringify(subtest));
+                    st.end();
+                    return;
+                }
+            } else if (subtest.hasOwnProperty('match')) {
+                st.fail('Test is invalid: Expected relationship type "match" is specified without input URL "i_ent": ' + JSON.stringify(subtest));
+                st.end();
+                return;
+            }
+            var act_clean = cleanURL(subtest.i_url);
+            if (subtest.clean) {
+                st.equal(act_clean, subtest.clean, 'Clean up');
+                if (subtest.i_ent && subtest.hasOwnProperty('match') && previous_match_tests.indexOf(subtest.i_ent + '+' + subtest.clean) === -1) {
+                    doMatchSubtest(st, subtest.i_ent, subtest.clean, 'clean', subtest.match);
+                }
+                tested = true;
+            }
+            if (subtest.i_rel && !subtest.valid && !subtest.wrong) {
+                st.fail('Test is invalid: Input relationship type "i_rel" is specified without neither "valid" nor "wrong" array: ' + JSON.stringify(subtest));
+                st.end();
+                return;
+            }
+            if (subtest.valid && subtest.wrong) {
+                st.fail('Test is invalid: Both "valid" and "wrong" arrays are specified whereas only one can be. (One defines the other.): ' + JSON.stringify(subtest));
+                st.end();
+                return;
+            }
+            if (subtest.valid || subtest.wrong) {
+                var match = subtest.i_rel || subtest.match;
+                var clean = subtest.clean || act_clean;
+                if (!match) {
+                    st.fail('Test is invalid: Validation test can not run without neither "match" nor input relationship type "i_rel": ' + JSON.stringify(subtest));
+                    st.end();
+                    return;
+                }
+                var nb_tested_rules = 0;
+                var validation_results = _.reduce(LINK_TYPES[match], function(results, rel_uuid, ent_name) {
+                    var rule = validationRules[rel_uuid];
+                    var is_valid = rule ? rule(clean) || false : true;
+                    results[is_valid].splice(_.sortedIndex(results[is_valid], ent_name), 0, ent_name);
+                    nb_tested_rules += rule ? 1 : 0;
+                    return results;
+                }, {true: [], false: []});
+                if (nb_tested_rules === 0) {
+                    st.fail('Validation test is worthless: No validation rule has been actually tested: ' + JSON.stringify(subtest));
+                } else {
+                    st.deepEqual(validation_results[subtest.valid ? true : false], (subtest.valid || subtest.wrong).sort(),
```

Comment:
Yes. I initially implemented this with ResidentAdvisor reviews in mind, the complete file proves this was unneeded.